### PR TITLE
Enforced authorization checks on all endpoints

### LINKED ISSUE #76: Enforce authorization checks on all existing endpoints
For every existing endpoint, make sure authorization is checked properly
* For actions that require authentication, use the appropriate middleware (401 Unauthorized)
* Same for admins (403 Forbidden)
* For actions that can only be performed by specific users, check and return 403 Forbidden
  * Users can't view or edit other users' data

It may be the case that everything's fine already, in which case this ticket is a no-op :)

### PR DESCRIPTION
**Pull request by amberchow8**
Closes #76

### CODE REVIEW

**Comment by amberchow8** [comment]
I wasn't sure about some changes I added to the PUT /studies/:studyId/tasks/:taskId so I added comments in the code

**Review by ross3102 — CHANGES_REQUESTED** [review]
Great job overall, left comments about a some minor things I noticed mostly with 403 vs 404 being the right call for data the user doesn't have access to. For this type of stuff its prob also a good idea to add API tests for these endpoints to make sure the user can properly see/create/update the data they should be able to see/create/update (and get appropriate errors in other cases) if you have time! I don't remember the state of our testing infra though lol so there might not be patterns in place for e.g. authenticated tests?

Also in the future if there's stuff that you're not sure about that you want to call out in the PR, you can leave comments on the lines in question right on GitHub instead of in the code! (Just go to files changed and leave a single comment on a line as if you were reviewing your own PR). That way if everything's good you can just merge without deleting the comments, and also a reviewer can reply directly to the comment. You can also call out general questions in the PR description as well!

File: packages/api/src/routes/studies.ts
Comment (amberchow8) [reply packages/api/src/routes/studies.ts:41]: I couldn't find an easy way to populate the study found from the user without using the Study.findById because it's an object id from the user

File: packages/api/src/routes/studies.ts
Comment (amberchow8) [reply packages/api/src/routes/studies.ts:73]: I'm not sure if I implemented the logic correctly but I implemented logic based on the study existing in the DB and the study existing in the user

File: packages/api/src/routes/tasks.ts
Comment (amberchow8) [reply packages/api/src/routes/tasks.ts:48]: I couldn't figure out a way to simplify this endpoint. I tried getting the study with Study.findById but I can't access its batteries because it returns type Query

File: packages/api/src/routes/studies.ts
Comment (amberchow8) [reply packages/api/src/routes/studies.ts]: Updated with owner property in a Study

**Review by DavidOB1 — APPROVED** [review]
Should wait for Ross' approval doe

File: packages/api/src/routes/studies.ts
Comment (ross3102) [inline packages/api/src/routes/studies.ts:98]: I think there's a case here that's not covered:
* user 1:
  * study 1:
    * task 1
* user 2:
  * study 2:
    * task 2

user 1 hits `PUT /studies/1/tasks/2`, seems like they are able to bypass the security and update task 2 (i think) (probably worth adding a test for this case)

File: packages/api/src/routes/studies.ts
Comment (ross3102) [inline packages/api/src/routes/studies.ts:123]: like david mentioned i think we need to pass something (maybe e) to `next` in these cases, otherwise no error will be returned

File: packages/api/src/routes/tasks.ts
Comment (ross3102) [inline packages/api/src/routes/tasks.ts:28]: could probably make all of this one query instead of populating every study - not that familiar with mongoose but i think you can do something like `Study.exists({owner: user._id, batteries: { $contains: req.params["id"] } })`

File: packages/api/src/routes/studies.ts
Comment (ross3102) [inline packages/api/src/routes/studies.ts:57]: can't leave a comment on unchanged code but we'll need to set owner to user._id on line 60 here

File: packages/api/src/routes/tasks.ts
Comment (ross3102) [reply packages/api/src/routes/tasks.ts:48]: oh just saw this, you just have to `await` Querys to get the data

File: packages/api/src/models/user.ts
Comment (ross3102) [inline packages/api/src/models/user.ts:13]: don't need this export

**Review by ross3102 — APPROVED** [review]
LGTM! just one extra export in there

**Pull request merged**
